### PR TITLE
add stack tag capabilities

### PR DESCRIPTION
**Pull request by drmaas**
* allow stack updates and creates to specify tags
* get stack returns tags
* filter stack list by query params

### CODE REVIEW

File: core/src/main/java/org/openstack4j/model/heat/builder/StackCreateBuilder.java [hunk 1]
```diff
@@ -76,5 +76,12 @@
     * @return StackCreateBuilder
     */
     StackCreateBuilder files(Map<String,String> files);
+
```
Comment (emjburns) [inline core/src/main/java/org/openstack4j/model/heat/builder/StackCreateBuilder.java]: Funny indentation here

File: core/src/main/java/org/openstack4j/openstack/heat/domain/HeatStack.java [hunk 2]
```diff
@@ -47,6 +47,8 @@
 	private List<GenericLink> links;
 	@JsonProperty("updated_time")
 	private String updatedTime;
+	@JsonProperty("tags")
```
Comment (emjburns) [inline core/src/main/java/org/openstack4j/openstack/heat/domain/HeatStack.java]: you shouldn't need this if the variable name is the same as the json name

File: core/src/main/java/org/openstack4j/model/heat/builder/StackCreateBuilder.java (hunk 1 above)
Comment (drmaas) [inline core/src/main/java/org/openstack4j/model/heat/builder/StackCreateBuilder.java]: will reformat whole file, it was a bit odd.

File: core/src/main/java/org/openstack4j/openstack/heat/domain/HeatStack.java (hunk 2 above)
Comment (drmaas) [inline core/src/main/java/org/openstack4j/openstack/heat/domain/HeatStack.java]: Looks like it is convention to be explicit.

**Comment by drmaas** [comment]
@gondor

File: core/src/main/java/org/openstack4j/openstack/heat/domain/HeatStack.java (hunk 2 above)
Comment (emjburns) [inline core/src/main/java/org/openstack4j/openstack/heat/domain/HeatStack.java]: Good to know. I've seen varying things across the project.

Comment (drmaas) [inline core/src/main/java/org/openstack4j/openstack/heat/domain/HeatStack.java]: Meaning, explicit for this class. Not sure about others.

**Comment by drmaas** [comment]
@vinodborole

**Comment by vinodborole** [comment]
LGTM

**Pull request merged**